src/bulk/common.ts

```diff
@@ -550,8 +530,8 @@ function executeCommands(
     }
 
     // Merge the results together
-    const writeResult = new BulkWriteResult(bulkOperation.s.bulkResult);
     const mergeResult = mergeBatchResults(batch, bulkOperation.s.bulkResult, err, result);
+    const writeResult = new BulkWriteResult(bulkOperation.s.bulkResult);
```

Comment:
`mergeBatchResults()` mutates the provided `bulkResult` so we move the construction of the `BulkWriteResult` until afterwards so it acts as immutable.

Reply:
Just clarification for further reviewers.  

Before the changes in this PR, `BulkWriteResult` used getters to provide access to each property on the raw bulk result, so constructing the `BulkWriteResult` before modifying the raw result was okay.

Now, we assign the properties directly in the constructor, so we need to create the `BulkWriteResult` after we process the raw result.